dm/master/server.go

```diff
@@ -1559,3 +1559,130 @@ func (s *Server) getSourceRespsAfterOperation(ctx context.Context, taskName stri
 	wg.Wait()
 	return sortCommonWorkerResults(sourceRespCh)
 }
+
+// ListMemberMaster list member master information
+func (s *Server) ListMemberMaster(ctx context.Context, req *pb.ListMemberRequest) (*pb.ListMemberMasterResponse, error) {
+	log.L().Info("", zap.Stringer("payload", req), zap.String("request", "ListMemberMaster"))
+	resp := &pb.ListMemberMasterResponse{}
+
+	isLeader, needForward := s.isLeaderAndNeedForward()
+	if !isLeader {
+		if needForward {
+			return s.leaderClient.ListMemberMaster(ctx, req)
+		}
+		return nil, terror.ErrMasterRequestIsNotForwardToLeader
+	}
+
+	memberList, err := s.etcdClient.MemberList(ctx)
+	if err != nil {
+		resp.Msg = errors.ErrorStack(err)
+		return resp, nil
+	}
+
+	etcdMembers := memberList.Members
+	members := make([]*pb.MemberMasterResponse, 0, len(etcdMembers))
+	client := http.Client{
+		Timeout: 1 * time.Second,
+	}
+
+	for _, etcdMember := range etcdMembers {
+		alive := true
+		_, err := client.Get(etcdMember.ClientURLs[0] + "/health")
+		if err != nil {
+			alive = false
+		}
+
+		members = append(members, &pb.MemberMasterResponse{
+			Name:       etcdMember.Name,
+			MemberID:   etcdMember.ID,
+			Alive:      alive,
+			ClientURLs: etcdMember.ClientURLs,
+			PeerURLs:   etcdMember.PeerURLs,
+		})
+	}
+
+	resp.Result = true
+	resp.Members = members
+	return resp, nil
+}
+
+// ListMemberWorker list member worker information
+func (s *Server) ListMemberWorker(ctx context.Context, req *pb.ListMemberRequest) (*pb.ListMemberWorkerResponse, error) {
+	log.L().Info("", zap.Stringer("payload", req), zap.String("request", "ListMemberWorker"))
+	resp := &pb.ListMemberWorkerResponse{}
+
+	isLeader, needForward := s.isLeaderAndNeedForward()
+	if !isLeader {
+		if needForward {
+			return s.leaderClient.ListMemberWorker(ctx, req)
+		}
+		return nil, terror.ErrMasterRequestIsNotForwardToLeader
+	}
+
+	workers := s.scheduler.GetWorker()
+	members := make([]*pb.MemberWorkerResponse, 0, len(workers))
+
+	for _, worker := range workers {
+		members = append(members, &pb.MemberWorkerResponse{
+			Name:   worker.BaseInfo().Name,
+			Addr:   worker.BaseInfo().Addr,
+			Stage:  string(worker.Stage()),
+			Source: worker.Bound().Source,
+		})
+	}
+
+	resp.Result = true
+	resp.Members = members
+	return resp, nil
+}
+
+// ListMemberLeader list member leader information
+func (s *Server) ListMemberLeader(ctx context.Context, req *pb.ListMemberRequest) (*pb.ListMemberLeaderResponse, error) {
+	log.L().Info("", zap.Stringer("payload", req), zap.String("request", "ListMemberLeader"))
+	resp := &pb.ListMemberLeaderResponse{}
+
+	isLeader, needForward := s.isLeaderAndNeedForward()
+	if !isLeader {
+		if needForward {
+			return s.leaderClient.ListMemberLeader(ctx, req)
+		}
+		return nil, terror.ErrMasterRequestIsNotForwardToLeader
+	}
+
+	memberList, err := s.etcdClient.MemberList(ctx)
+	if err != nil {
+		resp.Msg = errors.ErrorStack(err)
+		return resp, nil
+	}
+
+	_, leaderName, _, err := s.election.LeaderInfo(ctx)
```

Comment:
seems that the information returned by `LeaderInfo` is enough, don't need to query etcd again? and we can update the result of `LeaderInfo` if necessary